src/libraries/System.Text.Json/src/System/Text/Json/Serialization/IJsonOnDeserialized.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Specifies that the JSON type should have its <see cref="OnDeserialized"/> method called after deserialization occurs.
+    /// </summary>
+    /// <remarks>
+    /// This behavior is only supported on types representing JSON objects.
+    /// Types that have a custom converter or represent collections or values do not support this behavior.
```

Comment:
nit: is this more clear?

```suggestion
    /// Types that have a custom converter or represent collections or primitive values do not support this behavior.
```